### DISCUSSION TITLE
Request for Webhook Confirmation of Successful Build and App Store Push

### DISCUSSION THREAD

**Discussion by DeepakAphro121** [codemagic-ci-cd/codemagic-docs#2763] [Q&A] [answered]
Subject: Request for Webhook Confirmation of Successful Build and App Store Push

Dear CodeMagic Support Team,

I hope this message finds you well.

I am reaching out to request assistance in setting up a webhook for confirmation that the build I have requested via the API has been successfully pushed to the App Store, or if there have been any errors during the process.

Could you please let me know if there is a service available to achieve this, or if you have any alternative solutions you can recommend?

Thank you for your time and support. I look forward to your response.

**Comment by DeepakAphro121** [comment]
can you please give me solution for above its urgent for me

**Comment by dam-ease** [comment]
Hello @DeepakAphro121.
Please can you help with more information on your use-case?
If you would like to set up webhooks, you can check out the guide [here](https://docs.codemagic.io/yaml-running-builds/webhooks/)

**Comment by DeepakAphro121** [comment]
Thank you for your response.

We are looking to set up a webhook that directly notifies us once the process of pushing a build to the Apple App Store is complete. Specifically, we would like to receive notifications regarding whether the push was successful or if there were any errors.

Our goal is to display this information to our customers, including whether the app has been updated and the exact time of the update.

Could you please guide us on how to achieve this?

**Reply by dam-ease** [reply, marked as answer]
Hello @DeepakAphro121,
I'd like to get more clarity on your goal

> Our goal is to display this information to our customers, including whether the app has been updated and the exact time of the update.

I see you mentioned making use of API's to trigger builds, are you doing white-labeling and by customers, you mean clients?

You can enable slack/email notifications and also [report build status](https://docs.codemagic.io/yaml-publishing/post-publish/) in a post-publishing script, you will be notified if builds are failing in which case you will know if App Store publishing failed. If you have release notes, distributing to external beta review groups, then Post-processing will be triggered and if something happens there, Apple email you about these issues, so you still get notified. 

Another way to achieve this will be using the Build distribution to tester groups using YAML Configuration, you can checkout the guide here: https://docs.codemagic.io/yaml-publishing/tester-groups/ although this will only report the clients when the build is successful.

**Comment by DeepakAphro121** [comment]
Thanks a lot.
